src/Router.js

```diff
@@ -110,12 +110,19 @@ export default class Router {
   }
 
   /**
-   * @param {function} fn - Function to execute while popstate listener is disabled
+   * @param {function} fn - Async function to execute w/o popstate listener
    */
   executeWithoutPopstateListener(fn) {
     this.__shouldHandlePopstateEvents = false;
-    fn();
-    this.__shouldHandlePopstateEvents = true;
+
+    // Execute the function, then re-enable the popstate listener
+    return fn().then((result) => {
```

Comment:
Why not just use `.finally`? It requires the consumer to pass an ES6 Promise (not a Deferred) but I think thats OK/preferred.

```js
return fn().finally(function() {
  this.__shouldHandlePopstateEvents = true;
});
```

Reply:
Good idea! I can add that in instead.

Reply:
Okay just added in this change

Reply:
Actually, had to switch back to the original setup (`.then(resolveHander, rejectHandler)`). `.finally` was giving CI some trouble since it requires polyfill.

Reply:
@jamesopti I thought of that too (using `.finally`) but we might impact other companies that are using nuclear-router without an ES7-spec Promise polyfill that ensures `.finally` is available.